src/architecture/architect.js

```diff
@@ -274,7 +274,7 @@ const architect = {
       layer_sizes = layer_sizes_and_options;
       options = {};
     } else {
-      layer_sizes = layer_sizes_and_options.slice(layer_sizes_and_options.length - 1);
+      layer_sizes = layer_sizes_and_options.slice(0, -1);
```

Comment:
```suggestion
      layer_sizes = layer_sizes_and_options.pop();
```
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop)
I think this is a simpler solution.